### PR TITLE
Adds API documentation 

### PR DESCRIPTION
**Pull request by rheone**
- Adds documentation to be build via [Sphynx](https://docs.readthedocs.io/en/stable/intro/getting-started-with-sphinx.html#using-markdown-with-sphinx) for automated [ReadTheDocs.io](https://readthedocs.io/)
   - Adds documentation project that exists as a store house (and execution point) for examples needed in the documentation 
 - Updates README.md adding badges, moving around links, making more sense of things
 - Adds basic [psake](https://github.com/psake/psake) build scripts

### CODE REVIEW

File: docs/Unsigned-Arithmetic-Operations.md [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+lhs:    13688493
+rhs:    57024
+result: 13745517
+```
+
+## Subtraction
+
+Both the `ByteArrayUtils.SubtractUnsignedBigEndian` and `ByteArrayUtils.SubtractUnsignedLittleEndian` methods return the subtraction result of the provided `right` (minuend) and `left` (subtrahend) byte arrays accounting for the appropriate endiness of the input.
+
+If the operation would result in a negative value, given we're only dealing with unsigned integer operations, the execution will throw a `InvalidOperationException`.
```
Comment (ahsteele) [inline docs/Unsigned-Arithmetic-Operations.md]:
```suggestion
If the operation would result in a negative value, given we're only dealing with unsigned integer operations, the execution will throw an `InvalidOperationException`. 
```

File: docs/Unsigned-Arithmetic-Operations.md [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+
+Seemingly conspicuously absent are the `TrySubtractBigEndian` and `TrySubtractLittleEndian` equivalents of the `TrySumBigEndian` and `TrySumLittleEndian` methods. In actuality the various TrySum methods allow for a negative `delta` and therefore are functionally equivalent for safe subtraction. 
+
+## Comparison
+
+It is often not enough to simply compare the lengths of two arbitrary byte arrays to determine the equality or the largest / smallest unsigned integer value encoded as bytes.
+
+Both `ByteArrayUtils.CompareUnsignedBigEndian` and `ByteArrayUtils.CompareUnsignedLittleEndian` provide the ability to easily compare byte arrays as their unsigned integer values.
+
+The result is similar to that of `IComparer.Compare(left, right)`. The signed integer indicates the relative values of `left and 'right':
```
Comment (ahsteele) [inline docs/Unsigned-Arithmetic-Operations.md]:
```suggestion
The result is similar to that of `IComparer.Compare(left, right)`. The signed integer indicates the relative values of `left` and `right`:
```

File: docs/Endian-Byte-Enumerables.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Endian Byte Enumerables
+
+The `LittleEndianByteEnumerable` and `BigEndianByteEnumerable` gives access to more cleanly treat little-endian and big-endian byte arrays as enumerables in an expected indexable manner regardless of the underlying endianness ignoring `0x00` valued most significant bytes and managing indexing of the most significant byte at the 0th index.
+
+This topic will be further expounded upon at a later date. In the mean time please feel free to browse the source code available
```
Comment (ahsteele) [inline docs/Endian-Byte-Enumerables.md]:
```suggestion
This topic will be further expounded upon at a later date. In the meantime please feel free to browse the source code available:
```

File: docs/Concurent-Endian-Byte-Enumerables.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Concurent Endian Byte Enumerables
+
+`ConcurrentBigEndianByteEnumerable` and `ConcurrentLittleEndianByteEnumerable` allows for ease in parallel indexing a pair of byte arrays, that may not be of the same length, in the desired endianness. This comes in particularly useful when running bitwise or mathematical operations.
+
+This topic will be further expounded upon at a later date. In the mean time please feel free to browse the source code available
```
Comment (ahsteele) [inline docs/Concurent-Endian-Byte-Enumerables.md]:
```suggestion
This topic will be further expounded upon at a later date. In the meantime please feel free to browse the source code available:
```

File: docs/FixedBytes.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+`FixedBytes` implements 
+ - `IFormattable,`
+ - `IReadOnlyCollection<byte>`
+ - `IEquatable<FixedBytes>`
+ - `IEquatable<IEnumerable<byte>>`
+ - `IComparable<FixedBytes>`
+ - `IComparable<IEnumerable<byte>>`
+ - `IComparable`
+
+`FixedBytes` Overloads the operators
```
Comment (ahsteele) [inline docs/FixedBytes.md]:
```suggestion
Operator overloads provided by `FixedBytes`:
```

File: docs/Acknowledgements.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+- [Gulliver logo](https://commons.wikimedia.org/wiki/File:Gullivers_travels.jpg)
+  
+  > https://commons.wikimedia.org/wiki/File:Gullivers_travels.jpg
+  >
+  > This work is in the public domain in its country of origin and other countries and areas where the copyright term is the author's life plus 70 years or fewer. 
+  >
+  > U.S. work public domain in the U.S. for unspecified reason but presumably because it was published in the U.S. before 1924.
+  > This work has been identified as being free of known restrictions under copyright law, including all related and neighboring rights.
+
+## Thank Yous
```
Comment (nicksterx) [inline docs/Acknowledgements.md]: Do thank yous need to be here if they are empty?

File: .vscode/tasks.json [hunk 7]
```diff
@@ -0,0 +1,43 @@
+{
+   // See https://go.microsoft.com/fwlink/?LinkId=733558
+   // for the documentation about the tasks.json format
```
Comment (nicksterx) [inline .vscode/tasks.json]: // See https://github.com/psake/psake
// for more information on psake.ps1

File: README.md [hunk 8]
```diff
@@ -1,19 +1,22 @@
...
 
 Gulliver originally came to be for the sake of [Arcus]( https://github.com/sandialabs/arcus), a C# library for calculating, parsing, formatting, converting and comparing both IPv4 and IPv6 addresses and subnets. Inherently, by its nature, Arcus needed to do a great deal of byte manipulation. Eventually Gulliver came into a life of its own and it was decided that it should be broken off into its own library.
 
-
 ## Getting Started
 
-Gulliver is available via a [NuGet](https://www.nuget.org/packages/Gulliver/) package.
+Gulliver is available via a [NuGet](https://www.nuget.org/packages/Gulliver/). Our releases use [SemVer](http://semver.org/) for versioning.
+
+Documentation can be found on [Gulliver's Read the Docs site](https://arcus.readthedocs.io/en/latest/).
```
Comment (nicksterx) [inline README.md]: This link should point to https://gulliver.readthedocs.io/en/latest/

File: .vscode/tasks.json (hunk 7 above)
Comment (nicksterx) [inline .vscode/tasks.json]:
```suggestion
   // for the documentation about the tasks.json format
   // See https://github.com/psake/psake
   // for more information on psake.ps1
```

File: README.md (hunk 8 above)
Comment (nicksterx) [inline README.md]:
```suggestion
Documentation can be found on [Gulliver's Read the Docs site](https://gulliver.readthedocs.io/en/latest/).
```

File: docs/Acknowledgements.md (hunk 6 above)
Comment (nicksterx) [inline docs/Acknowledgements.md]:
```suggestion
```

No thank yours yet

**Comment by nicksterx** [comment]
Added My Review. Looks good to me.

**Pull request merged**